viam-cartographer.go

```diff
@@ -72,71 +75,54 @@ func SetDialMaxTimeoutSecForTesting(val int) {
 	dialMaxTimeoutSec = val
 }
 
-// TBD 05/04/2022: Needs more work once GRPC is included (future PR).
 func init() {
-	for _, slamLibrary := range slam.SLAMLibraries {
-		// TODO(PRODUCT-266): use triplet model names more properly here
-		sModel := resource.NewDefaultModel(resource.ModelName(slamLibrary.AlgoName))
-		registry.RegisterService(slam.Subtype, sModel, registry.Service{
-			Constructor: func(ctx context.Context, deps registry.Dependencies, c config.Service, logger golog.Logger) (interface{}, error) {
-				return New(ctx, deps, c, logger, false)
-			},
-		})
-		cType := slam.Subtype
-		config.RegisterServiceAttributeMapConverter(cType, sModel, func(attributes config.AttributeMap) (interface{}, error) {
-			var conf slamConfig.AttrConfig
-			decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Result: &conf})
+	registry.RegisterService(slam.Subtype, Model, registry.Service{Constructor: newCartographer})
+
+	config.RegisterServiceAttributeMapConverter(slam.Subtype, Model,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var attrCfg slamConfig.AttrConfig
+			decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Result: &attrCfg})
 			if err != nil {
 				return nil, err
 			}
 			if err := decoder.Decode(attributes); err != nil {
 				return nil, err
 			}
-			return &conf, nil
+			return &attrCfg, nil
 		}, &slamConfig.AttrConfig{})
-	}
 }
 
-// runtimeServiceValidation ensures the service's data processing and saving is valid for the mode and
+func newCartographer(ctx context.Context, deps registry.Dependencies, config config.Service, logger golog.Logger) (interface{}, error) {
```

Comment:
[optional] since this is a one-line function that's only used in one place, I think it would be cleaner to skip the helper function and just inline it